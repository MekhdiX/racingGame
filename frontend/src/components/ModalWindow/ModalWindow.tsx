import React from 'react';
import { Button, ButtonProps, Header, Icon, Modal } from 'semantic-ui-react';

type btnClick = (event: React.MouseEvent, data: ButtonProps) => void;

export const ModalWindow = (
  status: boolean,
  title: string,
  text: string,
  onCancel: btnClick | undefined,
  onOk: btnClick,
  saveButton: string,
  cancelButton?: string
): JSX.Element => (
  <Modal open={status}>
    <Header icon="archive" content={title} />
    <Modal.Content>
      <p>{text}</p>
    </Modal.Content>
    <Modal.Actions>
      {cancelButton !== undefined && onCancel !== undefined && (
        <Button onClick={onCancel} color="grey">
          <Icon name="remove" /> {cancelButton}
        </Button>
      )}
      <Button color="blue" onClick={onOk}>
        <Icon name="checkmark" /> {saveButton}
      </Button>
    </Modal.Actions>
  </Modal>
);
