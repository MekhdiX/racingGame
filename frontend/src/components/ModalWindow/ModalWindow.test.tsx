import ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';

import { ModalWindow } from './ModalWindow';

describe(`MainMenu Component`, () => {
  ReactDOM.createPortal = jest.fn((element) => {
    return element;
  }) as any;

  it(`renders correctly when open`, () => {
    const tree = renderer
      .create(
        ModalWindow(
          true,
          'Title',
          'Text',
          () => {},
          () => {},
          'Ok',
          'Cancel'
        )
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly when closed`, () => {
    const tree = renderer
      .create(
        ModalWindow(
          false,
          'Title',
          'Text',
          () => {},
          () => {},
          'Ok',
          'Cancel'
        )
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
