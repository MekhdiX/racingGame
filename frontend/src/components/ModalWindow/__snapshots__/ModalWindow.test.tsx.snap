// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu Component renders correctly when closed 1`] = `null`;

exports[`MainMenu Component renders correctly when open 1`] = `
<div
  className="ui page modals dimmer transition visible active"
>
  <div
    className="ui modal transition visible active"
    style={Object {}}
  >
    <div
      className="ui header"
    >
      <i
        aria-hidden="true"
        className="archive icon"
        onClick={[Function]}
      />
      <div
        className="content"
      >
        Title
      </div>
    </div>
    <div
      className="content"
    >
      <p>
        Text
      </p>
    </div>
    <div
      className="actions"
    >
      <button
        className="ui grey button"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="remove icon"
          onClick={[Function]}
        />
         
        Cancel
      </button>
      <button
        className="ui blue button"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="checkmark icon"
          onClick={[Function]}
        />
         
        Ok
      </button>
    </div>
  </div>
</div>
`;
